src/test/java/co/rsk/bitcoinj/script/ScriptTest.java

```diff
@@ -164,40 +146,48 @@ public void testIp() {
-        BtcTransaction transaction = PARAMS.getDefaultSerializer().makeTransaction(bytes);
-        TransactionOutput output = transaction.getOutput(1);
-        BtcTransaction spendTx = new BtcTransaction(PARAMS);
-        Address address = Address.fromBase58(PARAMS, "n3CFiCmBXVt5d3HXKQ15EFZyhPz4yj5F3H");
-        Script outputScript = ScriptBuilder.createOutputScript(address);
-        spendTx.addOutput(output.getValue(), outputScript);
-        spendTx.addInput(output);
+
+        Script multisigScript = ScriptBuilder.createMultiSigOutputScript(expectedNumberOfRequiredSignatures, federationKeys);
+
+        BtcTransaction fundingTx = new BtcTransaction(mainnetParams);
+        fundingTx.addInput(Sha256Hash.ZERO_HASH, 0, new Script(new byte[]{}));
+        fundingTx.addOutput(Coin.COIN, multisigScript);
+        TransactionOutput fundingTxOutput = fundingTx.getOutput(0);
+
+        BtcTransaction spendTx = new BtcTransaction(mainnetParams);
+        spendTx.addInput(fundingTxOutput);
+
+        Script recipientScript = ScriptBuilder.createOutputScript(userAddress);
+        Coin amountToSend = fundingTxOutput.getValue();
+        spendTx.addOutput(amountToSend, recipientScript);
+
         Sha256Hash sighash = spendTx.hashForSignature(0, multisigScript, SigHash.ALL, false);
-        BtcECKey.ECDSASignature party1Signature = key1.sign(sighash);
-        BtcECKey.ECDSASignature party2Signature = key2.sign(sighash);
-        TransactionSignature party1TransactionSignature = new TransactionSignature(party1Signature, SigHash.ALL, false);
-        TransactionSignature party2TransactionSignature = new TransactionSignature(party2Signature, SigHash.ALL, false);
+
+        List<TransactionSignature> signatures = new ArrayList<>();
+        for (int i = 0; i < expectedNumberOfRequiredSignatures; i++) {
+            BtcECKey.ECDSASignature partySignature = federationKeys.get(i).sign(sighash);
+            TransactionSignature partyTransactionSignature = new TransactionSignature(partySignature, SigHash.ALL, false);
+            signatures.add(partyTransactionSignature);
+        }
 
         // Create p2sh multisig input script
-        Script inputScript = ScriptBuilder.createP2SHMultiSigInputScript(ImmutableList.of(party1TransactionSignature, party2TransactionSignature), multisigScript);
+        Script inputScript = ScriptBuilder.createP2SHMultiSigInputScript(signatures, multisigScript);
 
         // Assert that the input script contains 4 chunks
-        assertEquals(4, inputScript.getChunks().size());
+        int expectedNumberOfChunksForScriptSig = 2 + expectedNumberOfRequiredSignatures;
```

Comment:
Could we use a constant for the number `2` to avoid magical numbers? WDYT?

Reply:
Updated.